internal/pritunl/user.go

```diff
@@ -26,10 +29,43 @@ type User struct {
 	OtpAuth         bool                     `json:"otp_auth,omitempty"`
 	DeviceAuth      bool                     `json:"device_auth,omitempty"`
 	Organization    string                   `json:"organization,omitempty"`
+	Pin             Pin                      `json:"pin,omitempty"`
 }
 
 type PortForwarding struct {
 	Dport    string `json:"dport"`
 	Protocol string `json:"protocol"`
 	Port     string `json:"port"`
 }
+
+type Pin struct {
+	IsSet  bool
+	Secret string
+}
+
+// MarshalJSON customizes the JSON encoding of the Pin struct.
+//
+// When marshaling a User JSON, the "pin" field will contain the PIN secret
+// if it is set, otherwise the field is excluded. This is used when making
+// a user create or update request to the Pritunl API.
+func (p Pin) MarshalJSON() ([]byte, error) {
```

Comment:
Add a pointer for a struct name in a receiver

Reply:
Changing the receiver from `Pin` to `*Pin` here breaks the existing User tests. `Pin` is a small struct so as I understand it there shouldn't be an advantage to using a pointer no?

Reply:
Actually took a better look at this and got it working alright. ✅